app/forms/waste_exemptions_engine/base_form.rb

```diff
@@ -17,38 +26,35 @@ def self.can_navigate_flexibly?
       true
     end
 
-    attr_accessor :token, :transient_registration
-
-    def initialize(registration)
-      # Get values from registration so form will be pre-filled
-      @transient_registration = registration
-      self.token = @transient_registration.token
+    def initialize(transient_registration)
+      # Get values from transient_registration so form will be pre-filled
+      @transient_registration = transient_registration
     end
 
-    def submit(attributes)
+    def submit(params)
+      attributes = transient_registration_attributes(params)
       attributes = strip_whitespace(attributes)
 
-      # Update the registration with params from the registration if valid
-      if valid?
-        @transient_registration.update_attributes(attributes)
-        @transient_registration.save!
-        true
-      else
-        false
+      attributes.each do |key, value|
+        transient_registration.public_send("#{key}=", value)
       end
-    end
 
-    # If the record is new, and not yet persisted (which it is when the start
-    # page is first submitted) then we have nothing to validate hence the check
-    validates :token, "defra_ruby/validators/token": true if @transient_registration&.persisted?
-    validate :registration_valid?
+      return transient_registration.save! if valid?
+
+      false
+    end
 
     private
 
+    def transient_registration_attributes(_params)
+      # Default behaviour. Subclasses to override.
+      {}
+    end
+
     def registration_valid?
```

Comment:
Do we really need this? I don't think we have any validations in place yet on the registration itself that is not re-defined in thee form. Let's discuss the possibility to drop this code.